azure-application-insights-spring-boot-starter/build.gradle

```diff
@@ -0,0 +1,51 @@
+/*
+ * ApplicationInsights-Java
+ * Copyright (c) Microsoft Corporation
+ * All rights reserved.
+ *
+ * MIT License
+ * Permission is hereby granted, free of charge, to any person obtaining a copy of this
+ * software and associated documentation files (the ""Software""), to deal in the Software
+ * without restriction, including without limitation the rights to use, copy, modify, merge,
+ * publish, distribute, sublicense, and/or sell copies of the Software, and to permit
+ * persons to whom the Software is furnished to do so, subject to the following conditions:
+ * The above copyright notice and this permission notice shall be included in all copies or
+ * substantial portions of the Software.
+ * THE SOFTWARE IS PROVIDED *AS IS*, WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED,
+ * INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR
+ * PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE
+ * FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR
+ * OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
+ * DEALINGS IN THE SOFTWARE.
+ */
+
+apply from: "$buildScriptsDir/common-java.gradle"
+apply from: "$buildScriptsDir/publishing.gradle"
+
+archivesBaseName = 'azure-application-insights-spring-boot-starter'
+
+dependencies {
+    compile (project(':core'))
+    compile (project(':web'))
+    provided('org.springframework.boot:spring-boot:1.5.9.RELEASE')
+    provided('org.springframework.boot:spring-boot-autoconfigure:1.5.9.RELEASE')
+    provided('org.springframework.boot:spring-boot-starter-web:1.5.9.RELEASE')
+    provided('org.springframework.boot:spring-boot-configuration-processor:1.5.9.RELEASE')
+
+    testCompile('org.junit.jupiter:junit-jupiter-api:5.0.2')
+    testCompile('org.junit.jupiter:junit-jupiter-params:5.0.2')
+    testCompile('org.junit.jupiter:junit-jupiter-engine:5.0.2')
+    testCompile('org.springframework.boot:spring-boot-starter-test:1.5.9.RELEASE')
+    testCompile('org.assertj:assertj-core:3.9.0')
+}
+
+// region Publishing properties
+
+projectPomName = project.msftAppInsightsJavaSdk + " Spring Boot starter"
```

Comment:
I took web module to copy this I intentionally did not include shadowJar because I have no need in it. 

ProjectPomName actually is equal to `Microsoft Application Insights Java SDK Spring Boot starter`, I don't know why other modules use `msftAppInsights` and then add `' Java SDK'`, for example `project.msftAppInsights + " Java SDK Core" == project.msftAppInsightsJavaSdk + ' Core'`

Reply:
Sure, shadow jar is not needed in this case. Shadowing is usually done only for dependencies that we need to encapsulate like Apache HTTP Client. In this case the SpringBoot Starter jar will be shipped as a jar and it will pull all the transitive dependencies once loaded into the SpringBoot Project, so we do not need that.